pcu/pcu_pmpi.c

```diff
@@ -23,12 +23,23 @@ void pcu_pmpi_init(MPI_Comm comm, pcu_mpi_t* self)
   MPI_Comm_dup(comm,&(self->coll_comm));
   MPI_Comm_size(comm,&(self->size));
   MPI_Comm_rank(comm,&(self->rank));
+  self->owned = 0;
 }
 
 void pcu_pmpi_finalize(pcu_mpi_t* self)
 {
   MPI_Comm_free(&(self->user_comm));
   MPI_Comm_free(&(self->coll_comm));
+  // Prevent accidental freeing of MPI_COMM_WORLD.
+  int result;
+  MPI_Comm_compare(self->original_comm, MPI_COMM_WORLD, &result);
```

Comment:
if you use `MPI_Comm_dup` above instead of directly using `MPI_COMM_WORLD` you will not need this.